src/main/java/Duke.java

```diff
@@ -1,10 +1,116 @@
+                shouldExit = true;
+                separator();
+                break;
+            }
+
+            if (command.equals("list")) {
+                list();
+            } else if (command.equals("done")) {
+                int taskId = Integer.parseInt(userInputArray[1]);
+                markAsDone(taskId);
+            } else if (command.equals("todo")) {
+                String todo = userInput.replace("todo", "").trim();
+                addTodo(todo);
+            } else if (command.equals("deadline")) {
+                int deadlineDateIndex = userInput.indexOf("/by");
+                String title = userInput.substring(0, deadlineDateIndex).replace("deadline", "");
+                String deadlineDate = userInput.substring(deadlineDateIndex+3, userInput.length());
+                addDeadline(title.trim(), deadlineDate.trim());
+            } else if (command.equals("event")) {
+                int atIndex = userInput.indexOf("/at");
+                String title = userInput.substring(0, atIndex).replace("event", "");
+                String atDate = userInput.substring(atIndex+3, userInput.length());
+                addEvent(title.trim(), atDate.trim());
+            }
+
+        }
+        exit();
+    }
+
+    public static void markAsDone(int taskId) {
+        Task selectedTask = items[taskId - 1];
+        selectedTask.markAsCompleted();
+        separator();
+        System.out.println("Nice! I've marked this task as done: ");
+        System.out.println(" " + selectedTask.getTitle());
+        separator();
+    }
+
+    public static void addTodo(String item) {
+        Task newTask = new Todo(item);
+        items[Task.numberOfTasks - 1] = newTask;
+        printSuccessfulAddMessage(newTask);
+    }
+
+    public static void addDeadline(String item, String deadline) {
+        Task newTask = new Deadline(item, deadline);
+        items[Task.numberOfTasks - 1] = newTask;
+        printSuccessfulAddMessage(newTask);
+    }
+
+    public static void addEvent(String item, String at) {
+        Task newTask = new Event(item, at);
+        items[Task.numberOfTasks - 1] = newTask;
+        printSuccessfulAddMessage(newTask);
+    }
+
+    public static void printSuccessfulAddMessage(Task newTask) {
+        separator();
+        System.out.println("Got it. I've added this task: ");
+        System.out.println("  " + newTask.getTitle());
+        System.out.println("Now you have " + Task.numberOfTasks + " tasks in the list.");
+        separator();
+    }
+
+    public static void list() {
+        int i = 0;
+        separator();
+        for(i = 0; i < Task.numberOfTasks; i++) {
+            Task currentTask = items[i];
+            System.out.println((i + 1) + ". " + currentTask.getTitle());
+        }
+        separator();
+    }
+
+    public static void echo(String command) {
+        separator();
+        System.out.println(command);
+        separator();
+    }
+
+    public static void separator() {
```

Comment:
Consider renaming this function to printSeparator() to follow the guideline that methods should be verbs.